pkg/apiserver/handlers.go

```diff
@@ -43,6 +42,12 @@ var specialVerbs = map[string]bool{
 	"watch":    true,
 }
 
+// TODO come up with a way to register these from storage (or other source of truth)
+// namespaceSubresouces is a set of all the subresources available on a namespace resource.  This is a special case because
+// URLs look like api/v1beta3/namspaces/<namespace name>/[subresource | resource].  We need to be able to distinguish the two
+// different cases.
+var namespaceSubresources = util.NewStringSet("status", "finalize")
```

Comment:
This looks like the wrong package for this. Can you make it part of the Strategy object(s)?